src/pages/ShoppingCart/ShoppingCart.js

```diff
@@ -0,0 +1,175 @@
+      ({ totalAmount, totalQuantity }, currentProduct) => {
+        return {
+          totalAmount:
+            totalAmount + currentProduct.cartAmount * currentProduct.data.price,
+          totalQuantity: totalQuantity + currentProduct.cartAmount,
+        };
+      },
+      { totalAmount: 0, totalQuantity: 0 }
+    );
+
+    setTotal(auxProductInfo.totalAmount);
+    setTotalProducts(auxProductInfo.totalQuantity);
+  }, [products, setTotalProducts]);
+
+  const updateQuantity = (event, productId) => {
+    const { value } = event.target;
+    const cartProduct = products.find((prod) => prod.id === productId);
+
+    if (cartProduct.data.stock < Number(value)) {
+      alert("Not enough stock for this product");
+    } else {
+      setProducts((prevProducts) =>
+        prevProducts.map((prod) =>
+          prod.id === productId ? { ...prod, cartAmount: Number(value) } : prod
+        )
+      );
+    }
+  };
+
+  const removeFromCart = (productId) => {
+    setProducts((prevProducts) =>
+      prevProducts.filter((prod) => prod.id !== productId)
+    );
+  };
+
+  return (
+    <ContentContainer>
+      <SectionContainer>
+        <CartTable>
+          <tbody>
+            <tr>
+              <CartTableNoBorderTd></CartTableNoBorderTd>
+              <CartTableHeaderTd>Name</CartTableHeaderTd>
+              <CartTableHeaderTd>Unit price</CartTableHeaderTd>
+              <CartTableHeaderTd>Quantity</CartTableHeaderTd>
+              <CartTableHeaderTd>Subtotal</CartTableHeaderTd>
+              <CartTableNoBorderTd></CartTableNoBorderTd>
+            </tr>
+            {products?.map((product) => (
+              <tr key={product.id}>
+                <CartTableTd>
+                  <CartProductImg src={product.data.mainimage.url} />
+                </CartTableTd>
+                <CartTableTd>{product.data.name}</CartTableTd>
+                <CartTableTd>${product.data.price}</CartTableTd>
+                <CartTableTd>
+                  <ProductQuantityInput
+                    type="number"
+                    value={product.cartAmount}
+                    min={0}
+                    max={product.data.stock}
+                    onChange={(e) => updateQuantity(e, product.id)}
+                  />
+                </CartTableTd>
+                <CartTableTd>
+                  ${product.data.price * product.cartAmount}
+                </CartTableTd>
+                <CartTableTd>
+                  <RemoveFromCartButton
+                    onClick={() => removeFromCart(product.id)}
+                  >
+                    <MdOutlineRemoveShoppingCart />
+                  </RemoveFromCartButton>
+                </CartTableTd>
+              </tr>
+            ))}
+            <tr>
+              <CartTableNoBorderTd></CartTableNoBorderTd>
+              <CartTableNoBorderTd></CartTableNoBorderTd>
+              <CartTableNoBorderTd></CartTableNoBorderTd>
+              <CartTableNoBorderTd></CartTableNoBorderTd>
```

Comment:
You could use the shorthand for this:

`<CartTableNoborderTd/>`

Reply:
Updated, thanks